### DISCUSSION TITLE
How can you pass an environment variable into a vercel-cli build command?

### DISCUSSION THREAD

**Discussion by dungle-scrubs** [vercel/vercel#9295] [Help] [answered]
Hi, I'm trying to pass a variable into a build with Vercel's latest CLI and although I see some documentation with an older CLI version (`vercel build --env`), I'm not having any luck with the current version.

For some context, I'm not attempting to pass any sensitive information, but rather just a simple boolean flag that I use in Next.js to avoid all static page generation for that specific build/deploy. Because we have hundreds of static pages, I'd like the ability to skip returning any paths from `getStaticPaths` for a very reduced build time, and I'd like to have the ability regardless of the environment.

**Comment by amyegan** [comment, marked as answer]
You can use [`vercel --env KEY1=value1 --env KEY2=value2`](https://vercel.com/docs/cli/deploy#env) to add or override environment variables for a specific deployment. Or you can use [`vercel env add`](https://vercel.com/docs/cli/env#usage) to add environment variables for all future deployments on a project.
